src/computed fields.js

```diff
@@ -0,0 +1,161 @@
+then the npc event's graphic automatically changes.
+
+This plugin is useful if the game has many state changes and/or many events reacting to state changes.
+
+The plugin field "computed-fields" lets you specify which fields can be computed.  See comments for
+this field at the bottom of this header.
+
+
+HOW TO USE - COMPUTE AVATAR'S GRAPHIC BASED ON WHICH SIDE OF THE ROOM THEY ARE ON
+1. Import this plugin into a game with an open room to move about in.
+2. Create two tiles for two avatar graphics.
+3. Create two tile fields on the Avatar event.  Name them "graphic-1" and "graphic-2".  Set them to the tiles of step 2.
+4. Change the avatar event's "graphic" field to be "javascript" type.  Set it to this code:
+  let f = "graphic-1";
+  if (EVENT.position[0] > 7) f = "graphic-2";
+  let data = FIELD(EVENT, f, "tile");
+  return { type: "tile", data: data };
+5. Run the game and move the player horizontally across the room.  Note that it's graphic changes halfway across.
+6. BONUS: Note that the avatar shows no graphic in the editor.  You can fix this by adding a "tile"
+		type field named "graphic" anywhere below the field of step 4.  The editor will use this
+		field for the graphic, while the playback will use the field of step 4 for the graphic.
+
+
+HOW TO USE - COMPUTE TERRAIN PASSABILITY BASED ON WHETHER AN ITEM IS TAKEN
+1. Import this plugin into a game with an open room to move about in.
+2. Draw a "boots" tile for an item.
+3. Draw a "spikes" tile for the floor.
+4. Add a "message" event beside the player.  Set the "say" field to "I put on the boots.".  Add a tag
+		field and name it "boots".  Add a tile field named "graphic" and set it to the boots tile of step 2.
+5. Add a "character" event a few cells from the avatar.  Set its graphic to the spikes tile of step 3.  Erase the "say" field.
+6. Change the "solid" tag field for the event of step 5 to javascript type.  Set it to this code:
+  if (!FIND_EVENT("boots")) return;
+  return { type: "tag", data: true };
+7. Copy the event of step 5.  Paste it around the player and the boots so that they are surrounded.
+8. Playtest the game.  Try to step over the spikes without getting the boots.  You should be blocked.  Walk over the boots and now the spikes should not block you.
+
+
+HOW TO USE - DIALOGUE CHANGES BASED ON HOW LONG THE GAME IS PLAYED
+1. Import this plugin into a game.
+2. Add a character event near the avatar with a custom graphic tile.
+3. Set the "say" field for the event of step 2 to a javascript type.  Set it to this code:
+  CLEAR_EVENT_SAY_CACHE(EVENT);
+  let data =
+    "We've been here for " +
+    PLAYBACK.frameCount +
+    " ticks.";
+  return { type: "dialogue", data: data };
+4. Run the game and repeatedly interact with the event of step 2.  Note that the dialogue changes.
+5. Also, not that the code begins with "CLEAR_EVENT_SAY_CACHE(EVENT);".  This is important to render
+	the latest "say" value.  Without it the dialogue is cached, so won't show changes.
```

Comment:
this isn't really a practical example of this plugin: the benefit of this plugin is being able to write declarative logic for state changes that would ordinarily require imperatively modifying events, but bipsi already lets you script dialogue in this way from the regular js fields by calling `SAY`.

i'd recommend removing all the `say`-specific parts of this plugin since it's redundant with the engine features